File: leetcode/cpp/26.remove-duplicates-from-sorted-array.cpp
class Solution {
public:
	int removeDuplicates(vector<int>& nums) {
		if (nums.empty()) return 0;
		int index = 0;
		for (size_t i = 1; i < nums.size(); i++)
		{
			if (nums[i] != nums[i - 1]) //不相同 指针右移
			{
				nums[++index] = nums[i];
			}
		}
		return index + 1;
	}
};

File: leetcode/rust/src/26.remove-duplicates-from-sorted-array.rs
impl Solution {
    pub fn remove_duplicates(nums: &mut Vec<i32>) -> i32 {
        let mut index  = 0;
        
        if nums.len() == 0{
            return 0i32;
        }
        
        for i in 1..(nums.len()){
            if nums[i] == nums[i-1]{
                // do nothing
            }
            else{
                index +=1;
                nums[index] = nums[i];
            }
        }
        (index + 1) as i32
    }
}
